conjureup/ui/widgets/step.py

```diff
@@ -170,6 +170,14 @@ def _build_form_fields(self):
                     select_w.select_option(i['default'])
                     field = StepField(key, label, select_w, i['type'])
                 else:
+                    is_yesno = input_type is YesNo
+                    is_intedit = input_type is IntegerEditor
+                    if value is not None and (
+                        (isinstance(value, bool) and not is_yesno) or
+                        (isinstance(value, int) and not is_intedit)
+                    ):
```

Comment:
Do we need to worry about other types here? e.g. `default=2.5` resulting in a `float`?

That's my only concern, otherwise this looks good.

Reply:
Float values don't have their own input type, so I'm assuming we don't support them at all and they'd be to be configured as `type: string` and `default: "2.5"`.

Reply:
Aye, `default: 2.5` is not supported, but we're trying to provide a more descriptive user when the user does something wrong, right?

If I'm reading this right, if they mistakenly put `default: 2.5`, then `value` will be of type `float` and this if condition will evaluate to `False`. Possibly leading to an unfriendly error like `'float' object has no attribute 'decode'` or whatever.